src/javascript-node-translation-info-factory.js

```diff
@@ -0,0 +1,35 @@
+const Pofile                = require('pofile');
+const { MARKER_NO_CONTEXT } = require('./constants.js');
+
+function toString(withLineNumbers = false) {
+  return (withLineNumbers && this.line)
+    ? `${ this.file }:${ this.line }`
+    : this.file;
+}
+
+function toPoItem(withLineNumbers = false) {
+  let poItem = new Pofile.Item();
+
+  poItem.msgid      = this.text;
+  poItem.references = [ this.reference.toString(withLineNumbers) ];
+  poItem.msgstr     = [];
+
+  return poItem;
+}
+
+function getNodeTranslationInfo(filename, localizedString, lineNumber) {
+  return {
+    text: localizedString,
+    reference: {
+      file: filename,
+      line: lineNumber,
+      toString,
+    },
+    context: MARKER_NO_CONTEXT,
+    toPoItem,
+  };
+}
+
+module.exports = {
+  getNodeTranslationInfo,
+};
```

Comment:
I'm not entirely comfortable with the name of this file :) why don't you just put these functions in the ``javascript-extract.js`` code and export the `getNodeTranslationInfo` function for the tests ?

Reply:
Why not, but honestly I actually prefer to separate the code in a few services for sake of readability and ease of maintenance. However I'm not comfortable with the naming of this file too (I was not really inspired at the moment I've created it) ;)

What do you think about it? Do you have any suggestions on the way I could name it? In other case I'll merge it with `javascript-extract.js`

Reply:
I tried to think about it and couldn't find any "extremely good" name. To me that was the sign that maybe the file was separated for a reason of "forced modularization", more than "well this looks like a good and unified separation that has a proper name" - hence, my proposal to just merge the files together. At least this way `javascript-extract.js`` is a good name that represents "this file contains everything related to javascript extraction of tokens". Do you agree ?

Reply:
I did create another file because `getNodeTranslationInfo`'s job is to build up and to return an object representation of the string being extracted. I wanted javascript-extract to deal only with the parsing and the search of the gettext tokens. Too me it is important not to end up with a file having too much responsibilities.

However It is another point of view, but It is okay for me ;)

Reply:
Alright, whatever you please !